modules/WebcastPolls/resources/webcastPollsKalturaProxy.js

```diff
@@ -128,81 +128,131 @@
         },
         transmitVoteUpdate : function(metadataId, userId, selectedAnswer,pollId)
         {
+			console.log(">>>>>##", "transmitVoteUpdate");
             var _this = this;
             var defer = $.Deferred();
-
-            if (metadataId && userId && selectedAnswer) {
-                var updateMetadataRequest = {
-                    service: "metadata_metadata",
-                    action: "update",
-                    id: metadataId,
-                    xmlData: '<metadata><Answer>' + selectedAnswer + '</Answer><UserId>' + userId + '</UserId></metadata>',
-                };
-
-                _this.log("transmitting update of vote for poll with id '" + pollId + "'");
-                _this.getKClient().doRequest(updateMetadataRequest, function (result) {
-                    if (!_this.isErrorResponse(result)) {
-                        _this.log('successfully transmitted update of vote');
-                        defer.resolve({});
-                    } else {
-                        _this.log("rejecting request due to error from kaltura api server");
-                        defer.reject();
-                    }
-                }, false, function (reason) {
-                    _this.log("rejecting request due to error from kaltura api server with reason " + (reason ? JSON.stringify(reason) : ''));
-                    defer.reject();
-                });
-            }else {
-                _this.log("rejecting request due to missing required information from plugin");
-                defer.reject({});
-            }
-
-            return defer.promise();
+			// New polls API for voting
+			if (pollId && userId && selectedAnswer) {
+				var vote = {
+					"service": "poll_poll",
+					"action": "vote",
+					"pollId": pollId,
+					"answerIds": selectedAnswer, // 1
+					"userId": userId
+				};
+				_this.getKClient().doRequest(vote, function (result) {
+					defer.resolve({});
```

Comment:
what is the API call returned 200 but with error...
see                   if (!_this.isErrorResponse(result)) {